keps/sig-architecture/617-improve-kep-implementation/README.md

```diff
@@ -0,0 +1,240 @@
+move the process forward.
+
+Additionally, a group of people from sig-arch started a KEP reading group.  We
+found it very difficult to keep track of what was happening with KEPs.  We
+identified a number of specific suggestions to make consuming KEPs more
+friendly.  History can be found in [this thread](https://groups.google.com/forum/#!msg/kubernetes-sig-architecture/SZ7UcFuOtCI/PQlTeSCICAAJ).
+
+### Goals
+
+To make it easier for humans and automation to use, find, reference, and
+consume KEPs.
+
+### Non-Goals
+
+- API Review process
+- Feature request triage
+- Developer guide
+
+## Proposal
+
+This proposal is somewhat omnibus.  Individual items are detailed below.
+
+### 1) Standardize an intra-KEP denotation for "not resolved"
+
+There is a perception that "merge and iterate" can be used to avoid dissent.
+If one topic is under debate, but other parts are acceptable, it should be
+possible to merge a KEP and iterate on the debatable point.  However, without
+reading the entire history of PRs on a KEP, exactly what is "under debate" is
+not clear.
+
+Requirements:
+  - not an HTML comment (so it is visible in the rendered markdown)
+  - unambiguous
+  - can wrap large bodies of text
+
+Parts of a KEP that are being merged without consensus shall be denoted as
+follows:
+
+```
+<<[UNRESOLVED optional text]>>
+Stuff that is under debate.
+<<[/UNRESOLVED]>>
+```
+
+The "optional text" can include a list of names or other context, but should be
+short.
+
+### 2) Change the names of KEPs
+
+The date-format in the name is not very human friendly and doesn't do anything
+for machines.  Instead, we propose to use the issue number of the enhancement
+issue as the KEP number.  KEP names would now be:
+`1234-short-human-friendly-string`.  This makes them easier to reference and
+remember: "KEP 1234" is easier than "KEP 20190903".  We will not retroactively
+change existing KEPs, so as to not break external links to them.
+
+### 3) Move KEPs from flat-files to a directory structure
+
+It's sometimes useful to have secondary documents alongside a KEP, for example
+image files.  To make this cleaner, new KEPs will be created in their own
+sub-directories of the SIG-specific directories. As per the above, a KEPs
+directory would be named as `1234-short-human-friendly-string`.  Inside that
+directory, the body of the KEP would be `README.md`.  Supporting documents and
+images would go in that directory.
+
+As part of this update, we will move all existing KEPs into this new structure,
+and leave links to the new locations in the old files.  This will preserve
+in-bound links and tell users where to find the information.
+
+### 4) Metadata split from KEP text
+
+Today, KEP metadata is a header on the markdown file.  This proposal moves that
+metadata to a new `kep.yaml` in each KEP directory (see above).  This file will
+contain the same information that was previously in the YAML front-matter.
```

Comment:
My previous comment about the enhancement issue should probably end up in this new `kep.yaml`.